test/micro/org/openjdk/bench/java/lang/VectorBitConversion.java

```diff
@@ -0,0 +1,125 @@
+package org.openjdk.bench.java.lang;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+@Warmup(iterations = 3, time = 5)
+@Measurement(iterations = 4, time = 5)
+@Fork(2)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@State(Scope.Thread)
+@BenchmarkMode(Mode.Throughput)
+public class VectorBitConversion {
+    @Param({"2048"})
+    public int size;
+
+    @Param("0")
+    private int seed;
+    private Random r = new Random(seed);
+
+    private double[] doubles;
+    private float[] floats;
+    private int[] ints;
+    private long[] longs;
+    private double[] resultDoubles;
+    private float[] resultFloats;
+    private int[] resultInts;
+    private long[] resultLongs;
+
+    @Setup
+    public void init() {
+        resultDoubles = new double[size];
+        resultFloats = new float[size];
+        resultInts = new int[size];
+        resultLongs = new long[size];
+
+        doubles = new double[size];
+        for (int i = 0; i < size; i++) {
+            doubles[i] = r.nextDouble();
+        }
+        floats = new float[size];
+        for (int i = 0; i < size; i++) {
+            floats[i] = r.nextFloat();
+        }
+        ints = new int[size];
+        for (int i = 0; i < size; i++) {
+            ints[i] = r.nextInt();
+        }
+        longs = new long[size];
+        for (int i = 0; i < size; i++) {
+            longs[i] = r.nextLong();
+        }
+    }
+
+    @Benchmark
+    public long[] doubleToLongBits() {
```

Comment:
Would something like this be more concise - 
```
@Benchmark
public long[] doubleToLongBits() {
    for (int i = 0; i < doubles.length; i++) {
            resultLongs[i] = Double.doubleToLongBits(aDouble);
    }
    return resultLongs;
}
```

The loop should still get vectorized (if vectorizable).

Reply:
Maybe but there's a reason why I wrote these benchmark methods this way. Keeping each line doing one thing makes it easier to map each line to the assembly (e.g. `perfasm`) and related IR nodes (e.g. `PrintIdeal`). That IMO is more important than the conciseness of the benchmark. What do others think?